src/data/equiv/basic.lean

```diff
@@ -444,6 +444,13 @@ calc (α × (β ⊕ γ)) ≃ ((β ⊕ γ) × α)       : prod_comm _ _
 @[simp] theorem prod_sum_distrib_apply_right {α β γ} (a : α) (c : γ) :
    prod_sum_distrib α β γ (a, sum.inr c) = sum.inr (a, c) := rfl
 
+def sigma_prod_distrib {ι : Type*} (α : ι → Type*) (β : Type*) :
```

Comment:
Should we call this `sigma_prod` and also have `prod_sigma`?

Reply:
Usually distributivity lemmas have the equality the other way around. Then it will be consistent with `sum_mul_distrib`.

Reply:
@rwbarton What's your opinion on this issue?

Reply:
I wrote the equivalence this way because (although it's hard to tell in the case of Set/`Type`) the forward map is the one which exists in any category and even replacing `- x β` with any functor: colim F(A_i) -> F(colim A_i). You can sort of see this when I prove that this is a homeomorphism for topological spaces: the forward map (as defined here) is obviously continuous but to prove the reverse map is continuous it's easiest to show the forward map is open (after proving a bunch of stuff about open maps and `sigma`).

Of course it does not really matter in the end which way the equivalence points because it is an equivalence, but at least on paper I find it a useful habit to keep maps written in the "natural" direction.

Reply:
I did notice the existing `sum_prod_distrib` and `prod_sum_distrib` are written in the opposite direction. I don't really have a strong opinion about which way these equivalences go but I suppose for consistency they should all go the same way.